File: src/ext_event_dns_base.php
<?hh
<*HAVE_LIBEVENT_DNS_SUPPORT*>final class EventDNSBase
<*HAVE_LIBEVENT_DNS_SUPPORT*>{
<*HAVE_LIBEVENT_DNS_SUPPORT*>    const int OPTION_SEARCH = 1 ;
<*HAVE_LIBEVENT_DNS_SUPPORT*>    const int OPTION_NAMESERVERS = 2 ;
<*HAVE_LIBEVENT_DNS_SUPPORT*>    const int OPTION_MISC = 4 ;
<*HAVE_LIBEVENT_DNS_SUPPORT*>    const int OPTION_HOSTSFILE = 8 ;
<*HAVE_LIBEVENT_DNS_SUPPORT*>    const int OPTIONS_ALL = 15 ;
<*HAVE_LIBEVENT_DNS_SUPPORT*>    private ?resource $__internal_resource = null;
<*HAVE_LIBEVENT_DNS_SUPPORT*>    <<__Native>> function __construct(EventBase $base, bool $initialize): void;
<*HAVE_LIBEVENT_DNS_SUPPORT*>}

File: src/ext_event_dns_base.cpp
#include "ext.h"

#ifdef HAVE_LIBEVENT_DNS_SUPPORT
namespace HPHP {
    static void HHVM_METHOD(EventDNSBase, __construct, const Object &base, bool initialize) {
        InternalResourceData *event_base_resource_data = FETCH_RESOURCE(base, InternalResourceData, s_event_base);
        Resource resource = Resource(NEWOBJ(InternalResourceData(evdns_base_new((event_base_t *) event_base_resource_data->getInternalResourceData(), initialize))));
        SET_RESOURCE(this_, resource, s_event_dns_base);
    }

    void eventExtension::_initEventDNSBaseClass() {
        HHVM_ME(EventDNSBase, __construct);
    }
}
#endif